wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Pruner.kt

```diff
@@ -53,8 +53,9 @@ class Pruner(
   }
 
   private fun retainImports(protoFiles: List<ProtoFile>): List<ProtoFile> {
+    val schema = Schema(protoFiles)
     return protoFiles.map { protoFile ->
-      protoFile.retainImports(protoFiles)
+      protoFile.retainImports(schema)
```

Comment:
Could we do the transformation of `protoFiles` to `schema` in the `retainImports` method so the public API doesn't change?

Reply:
We definitely could. The main concern is we’d be doing the same transformation multiple times for each proto file, but I’m not sure how expensive that is and if it would matter.

Alternatively I could add an overload that takes the schema and have the existing api delegate to it to so the change is only additive instead of breaking?

Reply:
Yes, ok

Reply:
Thumbs up to adding an overload

Reply:
Added and removed the breaking change.